datafusion/core/tests/parquet/row_group_pruning.rs

```diff
@@ -83,59 +91,64 @@ async fn test_prune_verbose(
 
 #[tokio::test]
 async fn prune_timestamps_nanos() {
-    test_prune(
+    test_row_group_prune(
         Scenario::Timestamps,
         "SELECT * FROM t where nanos < to_timestamp('2020-01-02 01:01:11Z')",
         Some(0),
         Some(1),
+        Some(0),
         10,
     )
     .await;
```

Comment:
There are so many parameters here it is getting hard to read the tests I think (as you have three different constants that need to be remembered)

Maybe we can (as a follow on PR) make a more self documenting, something like

```suggestion
    RowGroupPruningTest::new()
        .with_scenario(Scenario::Timestamps)
        .with_query("SELECT * FROM t where nanos < to_timestamp('2020-01-02 01:01:11Z')"),
        .with_expected_errors(Some(0)),
        .with_pruned_by_stats(Some(1)),
        .with_pruned_by_bloom_filter(Some(0)),
        .with_expected_rows(10)
    )
    .await;
```

Reply:
File https://github.com/apache/arrow-datafusion/issues/8922, thx @alamb